### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by imjasonh**
https://github.com/google/go-containerregistry/issues/916 is closed, update deps to pick up latest docker dependency and un-pin golang.org/x/sys

After this lands I'll update lifecycle as well.

@natalieparellano

### CODE REVIEW

**Comment by imjasonh** [comment]
Also cc @jromero @ekcasey related to https://github.com/buildpacks/imgutil/issues/70

File: go.mod [hunk 1]
```diff
@@ -1,13 +1,13 @@
 module github.com/buildpacks/imgutil
```
[the remark this answers is not in the document]
Comment (natalieparellano) [reply go.mod:1]: @muse-dev ignore

[the remark this answers is not in the document]
Comment (natalieparellano) [reply go.mod:1]: @muse-dev ignore

**Review by natalieparellano — APPROVED** [review]

**Review by jabrown85 — APPROVED** [review]
I'm a bit worried the `github.com/docker/docker` dep upgrade may be problematic in lifecycle or pack. We've been pinned on this old version for quite a while..but I think this is worth trying out again and documenting problems if we do run across them.

**Comment by natalieparellano** [comment]
From my experimentation it seems that both pack and the lifecycle should be able to bump their docker/docker as well :)

@imjasonh thanks so much for the PR!

**Pull request merged**